routing/routing.go

```diff
@@ -140,10 +142,83 @@ type RouteFilter struct {
 	Index int
 }
 
+// LBMetrics contains metrics used by LB algorithms
+type LBMetrics struct {
+	inflightRequests int
```

Comment:
Hard to tell for sure without a benchmark but I suspect that using atomic would be faster then mutex, see 
* https://golang.org/pkg/sync/atomic/#AddInt32
* https://stackoverflow.com/questions/47445344/is-there-a-difference-in-go-between-a-counter-using-atomic-operations-and-one-us

Reply:
Yes, I though about using atomics and it is likely I will, but I wanted to do it once we test the algorithm with some shadow traffic. I don't think the impact will be huge although it might be meaningful. Let's see if I can allocate some time tomorrow.